### PR TITLE
dict comparison handling, to include python-style ==

### PR DESCRIPTION
**Pull request by ivanlyon**
Filled out dict() comparison functions.  Handled == as python styled equality operator between dicts.

### CODE REVIEW

File: batavia/types/Dict.js [hunk 1]
```diff
@@ -52,19 +52,30 @@ batavia.types.Dict = function() {
      **************************************************/
 
     Dict.prototype.__lt__ = function(other) {
-        return this.valueOf() < other;
+        throw new batavia.builtins.TypeError("unorderable types: dict() < dict()");
```
Comment (phildini) [inline batavia/types/Dict.js]: What if `other` is an int or string or...?

This function should check for types and return the correct error message.

Alternatively, if you'd rather fill out all the types right now, throw the `NotImplemented` error if the type isn't `dict`.

File: batavia/types/Dict.js [hunk 2]
```diff
@@ -52,19 +52,30 @@ batavia.types.Dict = function() {
...
     };
 
     Dict.prototype.__le__ = function(other) {
-        return this.valueOf() <= other;
+        throw new batavia.builtins.TypeError("unorderable types: dict() <= dict()");
```
Comment (phildini) [inline batavia/types/Dict.js]: see above.

File: batavia/types/Dict.js [hunk 3]
```diff
@@ -52,19 +52,30 @@ batavia.types.Dict = function() {
...
     };
 
     Dict.prototype.__eq__ = function(other) {
-        return this.valueOf() == other;
+        if (batavia.isinstance(other, [batavia.types.NoneType]) == true) return false;
```
Comment (phildini) [inline batavia/types/Dict.js]: See above about what should happen if the type is not a dict or None.

throwing the `NotImplemented` for the other types is fine here too.

**Comment by phildini** [comment]
Thanks for submitting this! It's a good improvement. I'd like a few more cases covered, either by correct behavior or `NotImplemented` before accepting.

File: batavia/types/Dict.js (hunk 3 above)
Comment (ivanlyon) [inline batavia/types/Dict.js]: robert in15 dict_window

Pythonic "==" appears to throw no errors when dict is involved.  I can change it regardless, if I receive confirmation.

File: batavia/types/Dict.js (hunk 2 above)
Comment (ivanlyon) [inline batavia/types/Dict.js]: Good catch, changed in my latest commit.

File: batavia/types/Dict.js (hunk 1 above)
Comment (ivanlyon) [inline batavia/types/Dict.js]: Good catch, changed in my latest commit.

**Comment by freakboy3742** [comment]
Unfortunately, it appears that due to the confusion during the sprints, we've merged a different version of the comparison operators for dict, contributed by another user. Apologies for the time waste!

**Pull request closed without merging**